### PR TITLE
[Model] Add LocateAnything support

### PR DESCRIPTION
**Pull request by hoobnn**
## Summary

- Add native vLLM registration and config support for `nvidia/LocateAnything-3B`.
- Implement the LocateAnything multimodal model path with MoonViT vision encoding, the HF-compatible `mlp1` projector, Qwen2 language model wiring, image placeholder replacement, and checkpoint weight mapping.
- Add an optional LocateAnything slow-mode grammar logits processor that constrains AR decoding inside `<box>...</box>` blocks to valid `None`, point, or bbox token patterns.
- Add focused tests for config parsing, model registry integration, image token accounting, processor placeholder contracts, projector structure, and slow grammar constraints.

## Why this is not duplicating an existing PR

I checked upstream open PRs and issues before opening this draft:

- `gh pr list --repo vllm-project/vllm --state open --search "LocateAnything"`
- `gh pr list --repo vllm-project/vllm --state open --search "nvidia LocateAnything 3B"`
- `gh pr list --repo vllm-project/vllm --state open --search "locateanything multimodal model"`
- `gh issue list --repo vllm-project/vllm --state open --search "LocateAnything"`
- `gh issue list --repo vllm-project/vllm --state open --search "nvidia LocateAnything"`

These searches returned no matching open PRs or issues.

## Notes

This adds the standard vLLM autoregressive generation path for LocateAnything. The model can emit structured `<ref>...<box>...</box>` outputs when callers preserve special tokens, for example with `SamplingParams(skip_special_tokens=False)`.

The optional slow grammar logits processor can be enabled with:

```python
LLM(
    model="nvidia/LocateAnything-3B",
    trust_remote_code=True,
    logits_processors=[
        "vllm.model_executor.models.locate_anything.LocateAnythingSlowGrammarLogitsProcessor",
    ],
)
```

This PR does not implement the model's HF-native `generation_mode="fast"` or `generation_mode="hybrid"` Parallel Box Decoding path.

## Test Plan

- `pre-commit run`
- `.venv/bin/python -m pytest tests/transformers_utils/test_locate_anything_config.py tests/models/multimodal/test_locate_anything_processing.py tests/models/multimodal/test_locate_anything_projector.py tests/models/multimodal/test_locate_anything_init.py tests/models/multimodal/test_locate_anything_logits_processor.py -q`
  - Result: `17 passed, 16 warnings`
- `.venv/bin/python -m pytest tests/models/multimodal/test_locate_anything_hf_processor.py -q`
  - Result: `3 skipped, 16 warnings` locally because the external HF processor load was unavailable in this environment.
- Remote WSL + RTX 4070 Ti validation:
  - `tests/transformers_utils/test_locate_anything_config.py tests/models/multimodal/test_locate_anything_processing.py tests/models/multimodal/test_locate_anything_projector.py tests/models/multimodal/test_locate_anything_init.py tests/models/multimodal/test_locate_anything_hf_processor.py`
  - Result: `14 passed, 19 warnings`
  - Multi-task end-to-end script covering object detection, multi-instance phrase grounding, single phrase grounding, scene text detection, text grounding, GUI region grounding, and pointing.
  - Result: `all_tasks_ok=7`

## AI Assistance

This change was developed with AI assistance. The human submitter should review every changed line and verify the relevant tests before marking the PR ready for review.

### CODE REVIEW

**Comment by hoobnn** [comment]
Closing in favor of a re-created PR with a cleaned-up branch name and commit history. Replacement PR will be linked here.

**Comment by hoobnn** [comment]
Replacement: #44182
